va/http.go

```diff
@@ -0,0 +1,424 @@
+package va
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/letsencrypt/boulder/core"
+	berrors "github.com/letsencrypt/boulder/errors"
+	"github.com/letsencrypt/boulder/probs"
+)
+
+func newHTTPClient() http.Client {
+	// Construct a one-off HTTP client with a custom transport.
+	return http.Client{
+		Transport: &http.Transport{
+			// We are talking to a client that does not yet have a certificate,
+			// so we accept a temporary, invalid one.
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			// We don't expect to make multiple requests to a client, so close
+			// connection immediately.
+			DisableKeepAlives: true,
+			// We don't want idle connections, but 0 means "unlimited," so we pick 1.
+			MaxIdleConns:        1,
+			IdleConnTimeout:     time.Second,
+			TLSHandshakeTimeout: 10 * time.Second,
+		},
+	}
+}
+
+// httpValidationURL constructs a URL for the given IP address, path and port
+// combination. The port is omittied from the URL if it is the default HTTP
+// port. It is only used for constructing initial HTTP-01 validation requests
+// and so does not need to support constructing a URL with an HTTPS scheme.
+func httpValidationURL(validationIP net.IP, path string, port int) *url.URL {
+	urlHost := validationIP.String()
+
+	// If the port is something other than the conventional HTTP port, put it in
+	// the URL.
+	if port != 80 {
+		urlHost = net.JoinHostPort(validationIP.String(), strconv.Itoa(port))
+	} else if validationIP.To4() == nil {
```

Comment:
I think it would be clearer to make this `else if` its own independent `if` statement.